projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+	if(ret == CN0411_SUCCESS)
+		printf("Autoset Gain Resistance completed!\n");
+	else
+		printf("Autoset Gain Resistance failed!\n");
+}
+
+/**
+ * Set DAC value via UART
+ *
+ * @param args - pointer to the arguments on the command line.
+ * @param cn0411_dev - The device structure.
+ * @return none
+*/
+void CN0411_cmd_set_dac(uint8_t *args, cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+	uint8_t *p = args;
+	float input_val;
+	char arg[5];
+
+	/* Check if this function gets an argument */
+	while (*(p = CN0411_find_argv(p)) != '\0') {
+		/* Save DAC value */
+		CN0411_get_argv(arg, p);
+	}
+	input_val = atof(arg);
+	if(input_val < 0 || input_val > 2.5) {
+		ret = CN0411_FAILURE;
+		printf("Input out of range!.\n");
+	} else {
+		cn0411_dev->v_dac = input_val;
+		ret = CN0411_DAC_set_value(cn0411_dev, cn0411_dev->v_dac);
+		printf("DAC value set to %s V.\n", arg);
+	}
+	if (ret != CN0411_SUCCESS) {
+		printf("Set DAC value failed!.\n");
+	}
+}
+
+/**
+ * Set gain resistance via UART
+ *
+ * @param args - pointer to the arguments on the command line.
+ * @param cn0411_dev - The device structure.
+ * @return none
+*/
+void CN0411_cmd_gain_res(uint8_t *args, cn0411_device *cn0411_dev)
+{
+
+	int32_t ret = 0;
+	uint8_t *p = args;
+	char arg[5];
+
+	/* Check if this function gets an argument */
+	while (*(p = CN0411_find_argv(p)) != '\0') {
+		/* Save gain resistor value */
+		CN0411_get_argv(arg, p);
+	}
+
+	if(!strcmp(arg, "20M")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_20M);
+		printf("Gain Resistor set to 20M.\n");
+	} else if(!strcmp(arg, "2M")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_2M);
+		printf("Gain Resistor set to 2M.\n");
+	} else if(!strcmp(arg, "200K")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_200K);
+		printf("Gain Resistor set to 200K.\n");
+	} else if(!strcmp(arg, "20K")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_20K);
+		printf("Gain Resistor set to 20K.\n");
+	} else if(!strcmp(arg, "2K")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_2K);
+		printf("Gain Resistor set to 2K.\n");
+	} else if(!strcmp(arg, "200")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_200);
+		printf("Gain Resistor set to 200.\n");
+	} else if(!strcmp(arg, "20")) {
+		ret = CN0411_set_gain_res(cn0411_dev, CH_GAIN_RES_20);
+		printf("Gain Resistor set to 20.\n");
+	} else {
+		printf("Invalid argument!\n");
+	}
+	if (ret != CN0411_SUCCESS) {
+		printf("Set gain resistor value failed!\n");
+	}
+}
+
+/**
+ * Set RTD resistance value via UART
+ *
+ * @param args - pointer to the arguments on the command line.
+ * @param cn0411_dev - The device structure.
+ * @return none
+*/
+
+void CN0411_cmd_rtd_val(uint8_t *args, cn0411_device *cn0411_dev)
+{
+	uint8_t *p = args;
+	float input_val;
+	char arg[6];
+
+	/* Check if this function gets an argument */
+	while (*(p = CN0411_find_argv(p)) != '\0') {
+		/* Save RTD value parameter */
+		CN0411_get_argv(arg, p);
+	}
+	input_val = atof(arg);
```

Comment:
if you only integer values you could use atoi and change input_val to an integer type